psydac/api/tests/test_post_processing.py

```diff
@@ -0,0 +1,231 @@
+    V2 = VectorFunctionSpace('V2', domain, kind='hcurl')
+    V3 = VectorFunctionSpace('V3', domain, kind='hdiv')
+
+    domainh = discretize(domain, filename=geometry_file)
+
+    V1h = discretize(V1, domainh, degree=[4, 3])
+    V2h = discretize(V2, domainh, degree=[[3, 2], [2, 3]])
+    V3h = discretize(V3, domainh, degree=[[1, 1], [1, 1]])
+
+    uh = FemField(V1h)
+    vh = FemField(V2h)
+    wh = FemField(V3h)
+
+    npts_per_cell = refinement + 1
+
+    grid = [refine_array_1d(V1h.breaks[i], refinement, remove_duplicates=False) for i in range(2)]
+
+    # Output Manager Initialization
+    output = OutputManager('space_example.yml', 'fields_example.h5')
+    output.add_spaces(V1h=V1h, V2h=V2h, V3=V3h)
+    output.set_static().export_fields(u=uh, v=vh)
+
+    uh_grids = []
+    vh_grids = []
+
+    for i in range(15):
+        uh.coeffs[:] = np.random.random(size=uh.coeffs[:].shape)
+
+        vh.coeffs[0][:] = np.random.random(size=vh.coeffs[0][:].shape)
+        vh.coeffs[1][:] = np.random.random(size=vh.coeffs[1][:].shape)
+
+        # Export to HDF5
+        output.add_snapshot(t=float(i), ts=i).export_fields(u=uh, v=vh, w=wh)
+
+        # Saving for comparisons
+        uh_grid = V1h.eval_fields(grid, uh, npts_per_cell=npts_per_cell)
+        vh_grid =  V2h.eval_fields(grid, vh, npts_per_cell=npts_per_cell)
+        vh_grid_x, vh_grid_y = vh_grid[0][0], vh_grid[0][1]
+        uh_grids.append(uh_grid)
+        vh_grids.append((vh_grid_x, vh_grid_y))
+
+    output.export_space_info()
+    # End of the "simulation"
+
+    # =================================================================================
+    # Part 2: Post Processing
+    # =================================================================================
+
+    post = PostProcessManager(geometry_filename=geometry_file,
+                              space_filename='space_example.yml',
+                              fields_filename='fields_example.h5')
+
+    post.reconstruct_scope()
+
+    V1h_new = post.spaces['V1h']
+    V2h_new = post.spaces['V2h']
+
+    for i in range(len(uh_grids)):
+        snapshot = post.fields[i]
+        u_new = snapshot['fields']['u']
+        v_new = snapshot['fields']['v']
+
+        uh_grid_new = V1h_new.eval_fields(grid, u_new, npts_per_cell=npts_per_cell)
+        vh_grid_new = V2h_new.eval_fields(grid, v_new, npts_per_cell=npts_per_cell)
+        vh_grid_x_new, vh_grid_y_new = vh_grid_new[0][0], vh_grid_new[0][1]
+
+        assert np.allclose(uh_grid_new, uh_grids[i])
+        assert np.allclose(vh_grid_x_new, vh_grids[i][0])
+        assert np.allclose(vh_grid_y_new, vh_grids[i][1])
+
+    post.export_to_vtk('example_None', grid, npts_per_cell=npts_per_cell, dt=None, u='u', v='v', w='w')
+    post.export_to_vtk('example_int', grid, npts_per_cell=npts_per_cell, dt=5, u='u', v='v', w='w')
+    post.export_to_vtk('example_list', grid, npts_per_cell=npts_per_cell, dt=[9, 5, 6, 3], u='u', v='v', w='w')
```

Comment:
Well, you could always look at the `vtk` files to see that they look right (we would have to change the file slightly to write recognizable data and not just random values) or in the case of the identity mappings, check that the `pushforward` function gives us the same values. In the `bent_pipe` case I don't see how it could be checked without looking that the `vtk` file.